aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -108,25 +112,19 @@ def _refresh(self):
             )
             return
 
-        self.tags_by_arn = build_tags_by_arn_cache()
-        self.missing_arns -= set(self.tags_by_arn.keys())
+        self.tags_by_arn, last_modified = get_cache_from_s3()
+        if self._is_expired(last_modified):
+            send_forwarder_internal_metrics("s3_cache_expired")
+            self.tags_by_arn = build_tags_by_arn_cache()
+            write_cache_to_s3(self.tags_by_arn)
```

Comment:
How do avoid multiple forwarders detecting the cache expiration and attempting to refresh the cache around the same time? I thought we have discussed this case.

Reply:
✅